### ISSUE TITLE
translate fails on object message

### ISSUE THREAD

**Issue by jtojnar** [contributte/translation#3]
Thanks for the quick fix of #2. I encountered another issue related to that:

`Translator::translate` accepts `mixed` message arguments. With `Kdyby\Translation`, we could use `Nette\Utils\Html`, which was useful since it allowed us to avoid double translation of strings we wanted to pass to method:

```
$form->addError(Html::el()->setText($this->translator->translate('messages.team.error.too_few_members', $minMembers, ['category' => $category])));
```

In this case, we can just do `addError(…, false)` but in other cases like `Control` caption, this is [not possible](https://github.com/nette/forms/blob/85467ba6e58fd3ffbe5a6fa7be037e00a822925e/src/Forms/Controls/BaseControl.php#L279) so we rely on `null` HTML element.

However, `Contributte\Translation` fails when object is passed to `translate` method:

```
Argument 1 passed to Contributte\Translation\Helpers::extractMessage() must be of the type string or null, object given, called in …/vendor/contributte/translation/src/Translator.php on line 345
```

(Before 0.2.1, it failed the same way as #2; now the failure has been moved few lines down to `extractMessage`.)

#### What should translating object do?

In Nette 2.x there were [no typehints](https://github.com/nette/utils/blob/v2.5.3/src/Utils/ITranslator.php) in `ITranslator::translate` so `Kdyby\Translation` could get away with returning the original object:

https://github.com/Kdyby/Translation/blob/6b0721c767a7be7f15b2fb13c529bea8536230aa/src/Translator.php#L147-L149

Now that it expects to return string, we do not have many sensible choices for semantics:

* stringify the object if it has `__toString` method, and return the result
* stringify and translate
* print the class name https://github.com/nette/forms/blob/v3.0.0/tests/Forms/Controls.translate().phpt#L20 :clown_face: 
* throw an exception
* add `NoTranslate` class that would be stringified and other classes would be handled in one of the aforementioned ways (exception)
* do nothing, users can create `Translator` subclass and override `translate` method that handles `NoTranslate` class
* allow users to hook into translator and override object handling

**Comment by aleswita** [comment]
- stringify the object if it has __toString method, and return the result
- stringify and translate
// if convert **Nette\Utils\Html** to string with some element, its no possibility to translete, because output will be some like **<span>message.to.translate</span>** and Latte escape the output
- throw an exception
// I prefer this solution + make some helper class like **ObjectTranslator**

**Comment by jtojnar** [comment]
I have tried to make the “`ObjectTranslator`” on my end but could not come up with a way to globally use it in my app. DI does not like it when there are multiple classes implementing `Nette\Localization\ITranslator`.

https://github.com/jtojnar/entries/commit/46d8de390041dbd8f1467b7a6825cc71fd81ef37#diff-8e49bdec6d12b47165a62f09087a75ffR51

Perhaps if the extension allowed to override the `Translator` class used in

https://github.com/contributte/translation/blob/50523ac863e241bcd70513000a8907dd53a180db/src/di/TranslationExtension.php#L111

the we could set it to the following on the app end:

```php
<?php

declare(strict_types=1);

namespace App\Translation;

use Contributte;

class Translator extends Contributte\Translation\Translator {
	public function translate($message, ...$parameters): string {
		if ($message instanceof NoTranslate) {
			return (string) $message;
		}

		return $this->translator->translate($message, ...$parameters);
	}
}
```

**Comment by aleswita** [comment]
Fixed in v0.3.0
